app-creator/client/src/redux/reducer.ts

```diff
@@ -111,19 +128,24 @@ export const reducer: Reducer<AppCreatorStore, AppCreatorAction | AnyAction> = (
         ...state,
         selectedTemplateID: action.payload.id,
       };
-    case SET_IMPORTING:
+    case SET_EVENT_TYPE:
       return {
         ...state,
-        eventType: action.payload.value ? EventType.importing : EventType.none,
+        eventType: action.payload.value
+          ? action.payload.eventType
+          : EventType.none,
```

Comment:
Are all events guaranteed to have a non empty `action.payload.value`? If not, eventType will be set to none, which might cause unwanted behaviors later on.

Reply:
The `value` argument is mandatory in the action and has a boolean type. We can also default it to true. 

```Typescript
export const setEventType = (
  eventType: EventType,
  value: boolean = true
): SetEventType => {
  return {
    type: SET_EVENT_TYPE,
    payload: {
      eventType,
      value,
    },
  };
};
```